core/src/main/java/fj/data/Either.java

```diff
@@ -509,6 +530,41 @@ public void foreachDoEffect(final Effect1<B> f) {
     public <X> Either<A, X> sequence(final Either<A, X> e) {
       return bind(Function.<B, Either<A, X>>constant(e));
     }
+    /**
+       * Traverse with function that produces List (non-determinism).
+       *
+       * @param f the function to traverse with
+       * @return An either after traversing through this projection.
+    */
+      public <C> List<Either<A, C>> traverseList(final F<B, List<C>> f) {
+          return isRight() ?
+                  f.f(value()).map(x -> Either.right(x)) :
+                  list(Either.<A, C>left(e.left().value()));
+      }
+
```

Comment:
@mperry both resides on two different classes, i.e RightProjection and LeftProjection. client has to specify which projection he has to traverse, e.g e.right().traverseList(z -> ...). I think we should have `traverseLeftList` and `traverseRightList` in Either class.